File: 01.计算机基础知识/05.设计模式/04.行为型模式/03.责任链模式/java/scut/hujie/chainOfResPattern/ErrorLogger.java
package scut.hujie.chainOfResPattern;

public class ErrorLogger extends AbstractLogger {
	public ErrorLogger(int level) {
		this.level = level;
	}

	@Override
	protected void write(String message) {
		System.out.println("Error Console::Logger: " + message);
	}
}

File: 01.计算机基础知识/05.设计模式/04.行为型模式/03.责任链模式/c++/ErrorLogger.cpp
#include "ErrorLogger.h"


ErrorLogger::ErrorLogger()
{
}

ErrorLogger::ErrorLogger(int level) :AbstractLogger(level)
{
}

void ErrorLogger::write(string message) {
	cout << "Error Console::Logger: " << message << endl;
}

ErrorLogger::~ErrorLogger()
{
}
